GliaWidgetsTests/Lib/Coordinator/UIViewControllerCoordinatorTests.swift

```diff
@@ -0,0 +1,19 @@
+@testable import GliaWidgets
+import XCTest
+
+class UIViewControllerCoordinatorTests: XCTestCase {
+    func test_coordinate() {
+        let c1: UIViewControllerCoordinator = .mock
+        let c2: UIViewControllerCoordinator = .mock
+
+        var vc = c1.coordinate(to: c2)
+
+        // Store
+        XCTAssertEqual([c2.id: c2], c1.children)
+        // Free on vc deallocation
+        vc = UIViewController()
+        DispatchQueue.main.async {
```

Comment:
Is this necessary? DispatchQueue.main.async is side effect which should be avoided in unit tests.

Reply:
Yes. without it, every n-th test would fail. Like 1 in 10 or 20 or something. All debugging I did indicated that the deallocation might not happen straight away (maybe ARC isn't sync?). But even waiting a little bit etc makes the test pass

Reply:
I think we need to look deeper into this. Let me know if you need help to find out what is the reason for this.

Reply:
Fixed here. Calling onDeinit manually to simulate viewcontroller deallocation call as you suggested